flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/table.scala

```diff
@@ -66,6 +66,15 @@ class Table(
   private val tableSchema: TableSchema = new TableSchema(
     logicalPlan.output.map(_.name).toArray,
     logicalPlan.output.map(_.resultType).toArray)
+  
+  // map to save the all windows information registered in the table
+  private[flink] var windowPool: Map[Expression, GroupWindow] = Map()
+
+  // get the windowPool from the existing table
```

Comment:
return type of this builder function is a Table, and parameter type is also a Table, so
this function does "withWindowPoolFromGivenTable", and comment as "set windowPool of the Table from an existing Table Instance" maybe more precisely
the short function name "withWindowPool" is also good for me